spec.md

```diff
@@ -2260,6 +2288,8 @@ The CO MUST implement the specified error recovery behavior when it encounters t
 | Volume published but is incompatible | 6 ALREADY_EXISTS | Indicates that a volume corresponding to the specified `volume_id` has already been published at the specified `target_path` but is incompatible with the specified `volume_capability` or `readonly` flag. | Caller MUST fix the arguments before retrying. |
 | Exceeds capabilities | 9 FAILED_PRECONDITION | Indicates that the CO has exceeded the volume's capabilities because the volume does not have MULTI_NODE capability. | Caller MAY choose to call `ValidateVolumeCapabilities` to validate the volume capabilities, or wait for the volume to be unpublished on the node. |
 | Staging target path not set | 9 FAILED_PRECONDITION | Indicates that `STAGE_UNSTAGE_VOLUME` capability is set but no `staging_target_path` was set. | Caller MUST make sure call to `NodeStageVolume` is made and returns success before retrying with valid `staging_target_path`. |
+| Volume staged with different volume_mount_group | 9 FAILED_PRECONDITION | Indicates that volume with specified `volume_id` was node staged using different `volume_mount_group` on this node and hence can not be node published. | Caller MUST make sure that `NodePublishVolume` is called with same `volume_mount_group` which was used in `NodeStageVolume`. |
+| Volume already published with different volume_mount_group | 9 FAILED_PRECONDITION | Indicates that the volume with specified `volume_id` was already node published using different `volume_mount_group` on this node and can not be node published. | Caller MUST ensure that `NodePublishVolume` is called with same `volume_mount_group` on all target paths. |
```

Comment:
(a) if this is case is to cover the scenario when node-stage is unimplemented, please say so explicitly.
(b) unclear whether this behavior is "optional", meaning that if some implementations support multiple publish calls w/ different groups are they still required to throw this error? from earlier review comments in the spec it seems like this was to be optionally supported, but the language here does not make that clear.

Reply:
I have added some language to clarify the intent - https://github.com/container-storage-interface/spec/pull/468/commits/94346e334a3c1c5e92261d85df9d79124a3a8abe#diff-bc6661da34ecae62fbe724bb93fd69b91a7f81143f2683a81163231de7e3b545R2267

Reply:
So I have gone back little bit on this one  and updated the wording to be same for both nodestage->nodepublish and nodepublish->nodepublish.  Originally, the wording said CO MUST use same volume_mount_group identifier with NodePublish which was used during NodeStage.   I have relaxed with following wording:


```
  // If Plugin does not support NodePublishVolume with different
  // volume_mount_group than the one used during NodeStageVolume
  // then Plugin MAY return FAILED_PRECONDITION error.
  // Similarly if SP does not support NodePublishVolume of same volume
  // on same node but with different volume_mount_group it MAY return
  // FAILED_PRECONDITION error.
```

The idea is - some plugins may not perform actual mount on NodeStage and they may only do some housekeeping during staging and hence if underlying driver supports it, multiple NodePublish of same volume can use different gid than that what was specified on NodeStage. It did not make sense to penalize a Plugin for implementing NodeStage.

Reply:
cc @bswartz ^